src/Fields/Fields.jl

```diff
@@ -51,4 +53,13 @@ end
 
 include("set!.jl")
 
+struct SumOfFields{N, F}
+    fields :: F
+    SumOfFields{N}(fields...) where N = new{N, typeof(fields)}(fields)
+end
+
+@inline getindex(s::SumOfFields{1}, i, j, k) = @inbounds s.fields[1][i, j, k]
+@inline getindex(s::SumOfFields{2}, i, j, k) = @inbounds s.fields[1][i, j, k] + s.fields[2][i, j, k]
+@inline getindex(s::SumOfFields{3}, i, j, k) = @inbounds s.fields[1][i, j, k] + s.fields[2][i, j, k] + s.fields[3][i, j, k]
```

Comment:
You can also try to add

```julia
@propagate_inbounds function getindex(s::SumOfFields{N}, i...) where N
    first = getindex(SumOfFields(s.fields[1], s.fields[2], s.fields[3]), i...)
    last = getindex(SumOfFields(s.fields[4:N]...), i...)
    return first + last
end
```

To make this easier we might want the constructor

```julia
SumOfFields{N}(fields::Tuple) where N = new{N, typeof(fields)}(fields)
```